doc/game.md

```diff
@@ -45,8 +45,34 @@ A very simple `BaseGame` implementation example can be seen below:
     }
 ```
 
+## Flutter Widgets and Game instances
+
+Since a Flame game is a widget itself, it is quite easy to use Flutter widgets and Flame game together. But to make it even easier, Flame provides a `mixin` called `HasWidgetsOverlay` which will enable any Flutter widget to be show on top of your game instance, this makes it very easy to create things like a pause menu, or an inventory screen for example.
+
+To use it, simple add the `HasWidgetsOverlay` `mixin` on your game class, by doing so, you will have two new methods available `addWidgetOverlay` and `removeWidgetOverlay`, like the name suggests, they can be used to add, or remove widgets overlay above your game, they can be used like as show below:
+
+```dart
+addWidgetOverlay(
+  "PauseMenu", // Your overlay identifier
+  Center(child:
+      Container(
+          width: 100,
+          height: 100,
+          color: const Color(0xFFFF0000),
+          child: const Center(child: const Text("Paused")),
+      ),
+  ) // Your widget, this can be any Flutter widget
+);
+
+removeWidgetOverlay("PauseMenu"); // Use the overlay identifier to remove the overlay
```

Comment:
the fact that you need an overlay identifier hints that you can have multiple overlays, but this doesn't cover how rendering works priority-wise then... could we add another paragraph for that case?

Reply:
Done!